### PR TITLE
fromℕ≤ and inject≤ changes (Fin data type).

### PR DESCRIPTION
**Pull request by xekoukou**

### CODE REVIEW

**Comment by xekoukou** [comment]
In my system, there are no unsolved metas.
Which version is travis using?
Checked with :
```
Agda version 2.6.0-16eb89f -- tag v2.6.0
Agda version 2.6.0-89adf8e -- branch release-2.6.0
```

**Comment by MatthewDaggitt** [comment]
Travis is using Agda-2.6.0 installed directly from cabal. I'm getting an unsolved meta for the proof `toℕ-fromℕ≤″`. It can be fixed by changing it to:
```agda
toℕ-fromℕ≤″ : ∀ {m n} (m<n : m ℕ.<″ n) → toℕ (fromℕ≤″ m m<n) ≡ m
toℕ-fromℕ≤″ {m} {n} m<″n = begin
  toℕ (fromℕ≤″ m m<″n) ≡⟨ cong toℕ (sym (fromℕ≤≡fromℕ≤″ m<n m<″n)) ⟩
  toℕ (fromℕ≤ m<n)     ≡⟨ toℕ-fromℕ≤ m<n ⟩
  m                    ∎
  where m<n = ℕₚ.≤″⇒≤ m<″n; open ≡-Reasoning 
```

**Comment by MatthewDaggitt** [comment]
@xekoukou if you'd like this change in v1.1 any chance you could fix this in the next week or so?

**Comment by xekoukou** [comment]
Depends on #811

**Comment by MatthewDaggitt** [comment]
#811 has been merged in :+1:

**Comment by xekoukou** [comment]
@MatthewDaggit Done. (Didn't change anything.)

File: src/Data/Fin/Base.agda [hunk 1]
```diff
@@ -104,8 +105,8 @@ inject₁ zero    = zero
 inject₁ (suc i) = suc (inject₁ i)
 
 inject≤ : ∀ {m n} → Fin m → m ℕ.≤ n → Fin n
-inject≤ zero    (s≤s le) = zero
-inject≤ (suc i) (s≤s le) = suc (inject≤ i le)
+inject≤ {_} {suc n} zero    le = zero
+inject≤ {_} {suc n} (suc i) le = suc (inject≤ i (ℕₚ.≤-pred le))
```
Comment (gallais) [inline src/Data/Fin/Base.agda]: If these now irrelevant proofs of `m ℕ.≤ n` were marked as irrelevant,
we would get e.g. that `inject≤-irrelevant` becomes trivial.

Comment (xekoukou) [reply src/Data/Fin/Base.agda]: @gallais 
Making the proofs irrelevant introduce unsolved metas. Do you still want me to make them irrelevant? I will need to make changes to a few proofs.

Comment (xekoukou) [reply src/Data/Fin/Base.agda]: In my opinion, the fact agda couldn't solve those metas, while before it did, should be considered a bug.

Comment (xekoukou) [reply src/Data/Fin/Base.agda]: I made more arguments irrelevant and the metas were gone.

File: src/Data/Fin/Properties.agda [hunk 2]
```diff
@@ -364,14 +364,24 @@ lower₁-irrelevant {suc n} (suc i)  _   _ =
...
+                     → inject≤ i m≤n₁ ≡ inject≤ i m≤n₂
+inject≤-irrelevant {_} {suc n} zero m≤n₁ m≤n₂    = refl
+inject≤-irrelevant {_} {suc n} (suc i) m≤n₁ m≤n₂ = refl
+
+inject≤-idempotent : ∀ {m n k} (i : Fin m) (m≤n : m ℕ≤ n) (n≤k : n ℕ≤ k) (m≤k : m ℕ≤ k)
+                     → inject≤ (inject≤ i m≤n) n≤k ≡ inject≤ i m≤k
+inject≤-idempotent {_} {suc n} {suc k} zero _ _ _ = refl
+inject≤-idempotent (suc i) (s≤s m≤n) (s≤s n≤k) (s≤s m≤k) = cong suc (inject≤-idempotent i m≤n n≤k m≤k)
 
 ------------------------------------------------------------------------
```
Comment (gallais) [inline src/Data/Fin/Properties.agda]: This ought to work too:

```agda
inject≤-irrelevant : ∀ {m n} (i : Fin m) (m≤n₁ m≤n₂ : m ℕ≤ n)
                     → inject≤ i m≤n₁ ≡ inject≤ i m≤n₂
inject≤-irrelevant _ _ _ = refl

inject≤-idempotent : ∀ {m n k} (i : Fin m) .(m≤n : m ℕ≤ n) .(n≤k : n ℕ≤ k) .(m≤k : m ℕ≤ k)
                     → inject≤ (inject≤ i m≤n) n≤k ≡ inject≤ i m≤k
inject≤-idempotent {_} {suc n} {suc k}     zero    _ _ _ = refl
inject≤-idempotent {n = suc n} {k = suc k} (suc i) _ _ _ =
  cong suc (inject≤-idempotent i _ _ _)
```

Comment (xekoukou) [reply src/Data/Fin/Properties.agda]: Sorry, that was rather silly of me, not to notice that ```inject\le-irrelevant``` is not necessary at all.
I do not spend too much time on this. I hope it is merged soon.

**Comment by andreasabel** [comment]
Are we fine now with using irrelevance in the standard-library?  I thought the standard-library was conservative: no irrelevance, no instance arguments, no K, ...

**Pull request merged**